src/test/java/org/example/UserTest.java

```diff
@@ -50,78 +58,94 @@ public void getNonExistentUser() throws IOException {
     public void patchUser() throws IOException {
-        CloseableHttpClient httpclient = HttpClients.createDefault();
-        HttpPatch httpPut = new HttpPatch("https://gorest.co.in/public/v2/users/7881661?access-token=5d034ab1ba63642fd87b6ff1133eb5fbadad18da23c7c44bbe1f857e25b733b7");
+        CloseableHttpClient client = HttpClients.createDefault();
+        HttpPatch httpPatch = new HttpPatch(String.format("https://gorest.co.in/public/v2/users/7887693?access-token=%s", token));
         List<NameValuePair> params = new ArrayList<NameValuePair>(2);
         params.add(new BasicNameValuePair("name", "John Doe"));
-        params.add(new BasicNameValuePair("email", "mail@gmail.example"));
+        params.add(new BasicNameValuePair("email", String.format("%s@gmail.com", RandomStringUtils.random(10, true, true))));
         params.add(new BasicNameValuePair("gender", "male"));
         params.add(new BasicNameValuePair("status", "active"));
-        httpPut.setEntity(new UrlEncodedFormEntity(params));
-        HttpResponse response = httpclient.execute(httpPut);
-        Assert.assertEquals(response.getCode(), 200);
+        httpPatch.setEntity(new UrlEncodedFormEntity(params));
+        Header header = new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        httpPatch.setHeader(header);
+        String response = client.execute(httpPatch, new BasicHttpClientResponseHandler());
+        List<ValidationMessage> errors = validatePayload(response, "patchUserTemplate");
+        Assert.assertTrue(errors.isEmpty());
     }
 
     @Test
     public void putUser() throws IOException {
-        CloseableHttpClient httpclient = HttpClients.createDefault();
-        HttpPut httpPut = new HttpPut("https://gorest.co.in/public/v2/users/7824134?access-token=5d034ab1ba63642fd87b6ff1133eb5fbadad18da23c7c44bbe1f857e25b733b7");
+        CloseableHttpClient client = HttpClients.createDefault();
+        HttpPut httpPut = new HttpPut(String.format("https://gorest.co.in/public/v2/users/7887693?access-token=%s", token));
         List<NameValuePair> params = new ArrayList<NameValuePair>(2);
         params.add(new BasicNameValuePair("name", "John Doe"));
-        params.add(new BasicNameValuePair("email", "mail@gmail.example"));
+        params.add(new BasicNameValuePair("email", String.format("%s@gmail.com", RandomStringUtils.random(10, true, true))));
         params.add(new BasicNameValuePair("gender", "male"));
         params.add(new BasicNameValuePair("status", "active"));
         httpPut.setEntity(new UrlEncodedFormEntity(params));
-        HttpResponse response = httpclient.execute(httpPut);
-        Assert.assertEquals(response.getCode(), 200);
+        Header header = new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        httpPut.setHeader(header);
+        String response = client.execute(httpPut, new BasicHttpClientResponseHandler());
+        List<ValidationMessage> errors = validatePayload(response, "putUserTemplate");
```

Comment:
No Separation of Concerns
Tests mix HTTP logic, request creation, execution, and validation.

Reply:
You should break it into the following layers:

UserService – HTTP logic abstraction.

UserModel – POJO to represent the response body.

TestDataFactory – Creates consistent test data.

BaseTest – Reusable setup (e.g. token, client). 

Tests – Clean, readable, independent tests.